File: back end/rcf-soluciones-informaticas/orders-service/src/main/java/com/rcf/ordersservice/service/OrderStatusService.java
package com.rcf.ordersservice.service;

import com.rcf.ordersservice.dto.OrderStatusRequest;
import com.rcf.ordersservice.dto.OrderStatusResponse;

import java.util.List;

public interface OrderStatusService {
    List<OrderStatusResponse> getAllOrderStatuses();
    OrderStatusResponse getOrderStatusById(Long id);
    OrderStatusResponse saveOrderStatus(OrderStatusRequest orderStatusRequest);
    OrderStatusResponse updateOrderStatus(Long id, OrderStatusRequest orderStatusRequest);
    void deleteOrderStatus(Long id);
}

File: front end/rcf-soluciones-informaticas-angular/src/app/shared/services/order-services/order-status-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import OrderStatus from '../../model/OrderStatus';
import OrderStatusRequest from '../../model/Orders-Services/OrderStatusRequest.model';

@Injectable({
  providedIn: 'root'
})
export class OrderStatusService {
  private apiUrl = 'http://localhost:8080/orders'; 
  private domain = 'order-status';
  constructor(private http: HttpClient) {
  }  
  getOrderStatuses() : Observable<OrderStatus[]> {
    return this.http.get<OrderStatus[]>(`${this.apiUrl}/${this.domain}`);
  }
  getOrderStatusById(id: number) : Observable<OrderStatus> {
    return this.http.get<OrderStatus>(`${this.apiUrl}/${this.domain}/${id}`);
  }
  saveOrderStatus(orderStatus: OrderStatusRequest) : Observable<OrderStatus> {
    return this.http.post<OrderStatus>(`${this.apiUrl}/${this.domain}`, orderStatus);
  }
  updateOrderStatus(orderId: number, orderStatus: OrderStatusRequest) : Observable<OrderStatus> {
    return this.http.put<OrderStatus>(`${this.apiUrl}/${this.domain}/${orderId}`, orderStatus);
  }
  deleteOrderStatus(orderId: number) : Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${this.domain}/${orderId}`);
  }
}
